services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -1039,6 +1040,14 @@ public static Response finishedRequiredActions(KeycloakSession session, Authenti
         event.event(EventType.LOGIN);
         event.session(userSession);
         event.success();
+
+        UserModel authUser = userSession.getUser();
+        boolean isTemporaryAdmin = Boolean.parseBoolean(authUser.getFirstAttribute(TEMP_ADMIN_ATTR_NAME));
+        if (isTemporaryAdmin) {
+            logger.warnf("%s is a temporary admin user account. To harden security, " +
+                    "create a permanent admin account and delete the temporary one.", authUser.getUsername());
```

Comment:
It would be nice if we could just add an event listener and handle `LOGIN` and `CLIENT_LOGIN` events. It would avoid introducing code to those places to only add the warning while also keeping the code related to temporary admin user more isolated from the core code base.

Reply:
@pedroigor thanks for the input. I tried to come up with a solution as per your suggestion - [here are the changes](https://github.com/keycloak/keycloak/pull/31387/commits/9764da3c87b8552d0e96f88f96f96705d8f8065e) (in a separate commit for now).

I added a new event listener and I enabled it by default as we want these warnings to be clearly shown in the logs.
Alternatively, I could add the logic directly to the jboss-logging provider, which removes some portion of flexibility, but logically it might make sense.

cc: @vmuzikar

Reply:
Although I like the idea of leveraging an event listener, I'm not sure we should be adding a new separate listener just for this. It doesn't seem really systematic approach – what if we have more warnings about it, what if I want to see the warnings in a different listener, etc.? I'd probably prefer putting it into the jboss-logging provider.

@pedroigor @Pepo48 WDYT?

Reply:
I think it is a systematic approach because we can leverage the `jboss-logging` provider, as suggested by @Pepo48, to add whatever message we want. Not sure what you mean by "see the warnings in a different listener" though.

So yeah, +1 to have it on the jboss-event-listener.